docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+```
+Selector 2:
+```yaml
+        - matchBinaries:
+          - operator: "In"
+            values:
+            - "/bin/cat"
+          matchArgs:
+          - index: 1
+            operator: "Equal"
+            values:
+            - "/etc/shadow"
+          matchNamespaces:
+          - namespace: Net
+            operator: In
+            values:
+            - "host_ns"
+```
+
+We have `[`Selector1 `OR` Selector2`]`. Inside each selector we have `filters`.
+Both selectors have 3 filters (i.e. `matchBinaries`, `matchArgs`, and
+`matchNamespaces`) with different arguments. Adding a `-` in the beginning of a
+filter will result in a new selector.
+
+So the previous CRD will match if:
+
+`[`binary == /bin/cat `AND` arg1 == /etc/shadow `AND` MntNs == host`]` `OR`
+`[`binary == /bin/cat `AND` arg1 == /etc/shadow `AND` NetNs is host`]`
+
+We can modify the previous example as follows:
+
+> Generate a kprobe event if `/etc/shadow` was opened by `/bin/cat` which has
+> host `Net` and `Mnt` namespace access
+
+```yaml
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "example_ns_2"
+spec:
+  kprobes:
+    - call: "fd_install"
+      syscall: false
+      args:
+        - index: 0
+          type: int
+        - index: 1
+          type: "file"
+      selectors:
+        - matchBinaries:
+          - operator: "In"
+            values:
+            - "/bin/cat"
+          matchArgs:
+          - index: 1
+            operator: "Equal"
+            values:
+            - "/etc/shadow"
+          matchNamespaces:
+          - namespace: Mnt
+            operator: In
+            values:
+            - "host_ns"
+          - namespace: Net
+            operator: In
+            values:
+            - "host_ns"
+```
+
+Here we have a single selector. This CRD will match if:
+
+`[`binary == /bin/cat `AND` arg1 == /etc/shadow `AND` `(`MntNs == host `AND`
+NetNs == host`)` `]`
+
+#### Capabilities filter
+
+Capabilities filters can be specified under the `matchCapabilities` field and
+provide filtering of calls based on Linux capabilities in the specific sets.
+
+An example syntax is:
+```yaml
+- matchCapabilities:
+  - type: Effective
+    operator: In
+    values:
+    - "CAP_CHOWN"
+    - "CAP_NET_RAW"
+```
+
+This will match if: [`Effective` capabilities contain `CAP_CHOWN`] OR
+[`Effective` capabilities contain `CAP_NET_RAW`]
+
+- `type` can be: `Effective`, `Inheritable`, or `Permitted`.
+- `operator` can be `In` or `NotIn`
+- `values` can be any supported capability. A list of all supported
+  capabilities can be found in `/usr/include/linux/capability.h`.
+
+**Limitations**
+
+1. There is no limit in the number of capabilities listed under `values`.
+2. Only one `type` field can be specified under `matchCapabilities`.
+
+#### Namespace changes filter
+
+Namespace changes filter can be specified under the `matchNamespaceChanges`
+field and provide filtering based on calls that are changing Linux namespaces.
```

Comment:
How do we keep track of changes? I guess we already know what namespaces the process has access to and we match against a "delta" whenever certain system calls are invoked? That's my guess but would be nice to double check and put like 2 sentence about the logic.

Reply:
Yes it seems it's the idea:
https://github.com/cilium/tetragon/blob/1a68bf2681e79ec57c37863b1e5b809e7be04413/bpf/process/pfilter.h#L206-L209

Reply:
I added the sentences about the logic!